src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.OSX.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Net.Sockets
+{
+    internal static partial class SocketPal
+    {
+        public const bool SupportsMultipleConnectAttempts = false;
+
+        public static void PrimeForNextConnectAttempt(int fileDescriptor, int socketAddressLen)
```

Comment:
You could make this a partial method, with:
```C#
private static partial void PrimeForNextConnectAttempt(int fileDescriptor, int socketAddressLen);
```
to the main definition, and then you just wouldn't include an implementation for OS X and the call site would evaporate.